src/components/HelpScoutBeacon/HelpScoutBeacon.js

```diff
@@ -95,35 +95,81 @@ const HelpOptIn = React.memo(function HelpOptIn({
   const { above } = useViewport()
   const expandedMode = above('medium')
   const [mode, setMode] = useState(CLOSED)
+  const [iframe, setIframe] = useState(null)
 
-  const handleClose = React.useCallback(() => setMode(CLOSED), [])
+  // open and close atomic actions
+  const handleClose = useCallback(() => {
+    setMode(CLOSING)
+    if (beaconReady) {
+      window.Beacon('close')
+    }
+  }, [beaconReady])
+  const handleOpen = useCallback(() => {
+    setMode(OPENING)
+    if (beaconReady) {
+      window.Beacon('open')
+    }
+  }, [beaconReady])
+  // toggle between states, when optedId and not
   const handleToggle = useCallback(() => {
-    if (mode !== OPENING && mode !== CLOSING) {
-      setMode(mode === CLOSED ? OPENING : CLOSING)
+    if (beaconReady) {
+      mode === CLOSED || mode === CLOSING ? handleOpen() : handleClose()
```

Comment:
Could we keep using `if` expression for these, and keep the ternary operators for cases where we need to use the value from the evaluated expression?

Reply:
Done in e177082; I think we can implicitly have this rule for not calling functions in ternaries unless we'll be using their return values.